src/metalnx-web/src/main/java/com/emc/metalnx/controller/MetadataController.java

```diff
@@ -255,7 +264,20 @@ public void exportSearchResultsToCSVFile(final HttpServletResponse response) thr
 	@RequestMapping(value = "/getMetadata/", method = RequestMethod.POST)
 	public String getMetadata(final Model model, final String path) throws DataGridException, FileNotFoundException {
 		logger.info("MetadataController getMetadata() starts !!");
-		List<DataGridMetadata> metadataList = metadataService.findMetadataValuesByPath(path);
+		
+		Predicate<DataGridMetadata> metadataFilter = avu -> true;
+		final List<String> attributeNamePrefixes = new ArrayList<>();
+
+		if (null != metadataMaskPrefixes && !metadataMaskPrefixes.isEmpty() &&
+			null != metadataMaskDelimiter && !metadataMaskDelimiter.isEmpty())
+		{
+			final String[] elements = metadataMaskPrefixes.split(Pattern.quote(metadataMaskDelimiter));
+			attributeNamePrefixes.addAll(Arrays.asList(elements));
+			metadataFilter = avu -> attributeNamePrefixes.stream()
+					.noneMatch(prefix -> avu.getAttribute().startsWith(prefix));
```

Comment:
Should we trim leading and trailing whitespace from the attribute names?

Reply:
i... think so?   i don't think imeta does that though...  let's investigate.

Reply:
After some discussion, we should not trim whitespace because the user may have added it on purpose.